<launch>
	<arg name="rate" default="30" />
	<arg name="use_nodelets" default="false" />
	<arg name="nodelet_manager_name" default="nodelet_manager" />
	<arg name="name_prefix" default="tritech_micron" />
	<arg name="name_postfix" default="" />
	
	<arg name="name" value="$(arg name_prefix)$(arg name_postfix)" />
	<arg name="pkg" value="tritech_micron" />
	<arg name="node_type" value="$(arg pkg)_node" />
	<arg name="nodelet_type" value="$(arg pkg)_nodelet" />

	<arg name="simulate" default="false" />
	<arg name="use_debug_mode" default="true" />
	<arg name="frame_id" default="/tritech_micron" />
	<arg name="port" default="/dev/ttyUSB1" />
	<arg name="num_bins" default="400" />
	<arg name="range" default="10.0" />
	<arg name="velocity_of_sound" default="1500.0" />
	<arg name="angle_step_size" default="crazy_low" />

	<node pkg="$(arg pkg)" type="$(arg node_type)" name="$(arg name)" output="screen" args="_rate:=$(arg rate) _use_debug_mode:=$(arg use_debug_mode) _simulate:=$(arg simulate) _frame_id:=$(arg frame_id) _port:=$(arg port) _num_bins:=$(arg num_bins) _range:=$(arg range) _velocity_of_sound:=$(arg velocity_of_sound) _angle_step_size:=$(arg angle_step_size)" >
	</node>

	<arg name="show_reconfigure" default="true" />
	<include file="$(find seabee3_launch)/dynamic_reconfigure.launch" >
		<arg name="enable" value="$(arg show_reconfigure)" />
		<arg name="name" value="$(arg name)" />
	</include>
	
	<arg name="enable_scan_line_converter" default="true" />
	<group if="$(arg enable_scan_line_converter)" >
	<arg name="show_image" default="true" />
	<include file="$(find tritech_micron)/launch/scan_line_converter.launch" >
		<arg name="rate" value="$(arg rate)" />
		<arg name="scan_line" value="$(arg name)/scan_line" />
		<arg name="show_image" value="$(arg show_image)" />
		<arg name="show_reconfigure" value="$(arg show_reconfigure)" />
	</include>
	</group>

	<arg name="show_rviz" default="true" />
	<include file="$(find seabee3_launch)/rviz.launch" >
		<arg name="enable" value="$(arg show_rviz)" />
		<arg name="use_config_file" value="true" />
		<arg name="config_file" value="$(find tritech_micron)/launch/tritech_micron.vcg" />
	</include>
</launch>
